### PR TITLE
[terra-button] Allow hover styles to apply only on non-touch devices.

### LINKED ISSUE #2891: [terra-button] Hover highlight enabled after click on mobile devices
# Bug Report

## Description
After clicking a button, the button displays as though it is hovered on mobile devices.

## Steps to Reproduce
This behavior is reproducible using the button"variant example on the [documentation page](https://engineering.cerner.com/terra-ui/components/terra-button/button/button) on mobile.
1. Click on Button with "Neutral" text on it.
2. Button remains highlighted with hover effect.

## Expected Behavior
If button is clicked on mobile device I would expect it to remove the hover effect after click.

## Environment
* Component Name and Version: terra-button
* Browser Name and Version:
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile): mobile

## @ Mentions
@StephenEsser

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
This PR resolves the issue by making use of hover and pointer interaction media features introduced in Media Queries Level 4 so that the hover styles are applied only on non-touch devices whose primary input is a fine pointer device such as a mouse or a stylus.
Closes #2891 

### Deployment Link
Examples: https://terra-core-deployed-pr-2929.herokuapp.com/components/terra-button/button/button
Tests: https://terra-core-deployed-pr-2929.herokuapp.com/tests/terra-button/button/button-modal-focus

### Additional Details
Updated wdio snapshots for the following components also due to changes in button-

- terra-button-group
- terra-paginator
- terra-search-field
- terra-toggle-button

Based on fix introduced for terra-list: #2332

### CODE REVIEW

File: packages/terra-button-group/src/ButtonGroup.module.scss [hunk 1]
```diff
@@ -30,17 +30,18 @@
       // A z-index needs to be added on focus so that the focus ring does not get hidden behind adjacent buttons
       z-index: 1000;
     }
+    @media screen and (-ms-high-contrast: active), (-ms-high-contrast: none), (hover: hover) and (pointer: fine) {
```
Comment (yuderekyu) [inline packages/terra-button-group/src/ButtonGroup.module.scss:33]: what happens if we don't include the ms-high-contrast queries?

Comment (saket2403) [reply packages/terra-button-group/src/ButtonGroup.module.scss:33]: The interaction media features hover and pointer are not supported on IE and it has been addressed by making use of -ms-high-contrast.

File: packages/terra-paginator/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ ChangeLog
 Unreleased
 ----------
 ### Changed
+* Updated wdio snapshots due to changes in button.
```
Comment (ShettyAkarsh) [inline packages/terra-paginator/CHANGELOG.md:7]: Make sure in framework and clinical repo's there are no wdio failures. by referring this branch over there

**Comment by neilpfeiffer** [comment]
Need to investigaet why byproduct of this fix does not work for screenshot tests - such that the hover tests no longer show the hover styles as intended by the tests -- one of the examples but applies to all the button variants:
https://github.com/cerner/terra-core/pull/2929/files?short_path=a840576#diff-a840576a87891549b705162e12692ebb

Looks like it also is a problem in list, where this fix was first applied. Although -- not sure why, it didn't seem to cause an update for screenshots when it was first applied on [Apr 10, 2019](https://github.com/cerner/terra-core/commits/master/packages/terra-list/src/List.module.scss) -- instead, it wasn't until [Dec 2, 2019](https://github.com/cerner/terra-core/commits/master/packages/terra-list/tests/wdio/__snapshots__/reference/en/chrome_medium/list-spec/ListItem_Test%5BListItem-Hover%5D.png) that the screenshots changed.

**Comment by neilpfeiffer** [comment]
Tested: PR #2929
Resolves: Issue #2891

**Reviewed:**
- Tested on the [Click>"Click Me" example](https://terra-core-deployed-pr-2929.herokuapp.com/components/terra-button/button/button) and [Button Modal Focus test](https://terra-core-deployed-pr-2929.herokuapp.com/tests/terra-button/button/button-modal-focus) on iOS: iPhone, to ensure hover style is being applied for mouse/pointer, but not applying and sticking for iOS.

\
\
**Will Need to Resolve:**
- Introduced this new wdio testing issue: #2947

which will also need to retro-fix the same problem for [hover tests in terra-list](https://github.com/cerner/terra-core/commits/master/packages/terra-list/tests/wdio/__snapshots__/reference/en/chrome_medium/list-spec/ListItem_Test%5BListItem-Hover%5D.png) introduced with #2332, and the same fix will need to be applied in PR #2932, introduced by companion issue #2927


\
\
Verified by @neilpfeiffer - approving conditionally that https://github.com/cerner/terra-core/pull/2929#issuecomment-620058376 will be resolved on followup issue.

**Pass** |
-----------|
UX Review|

**Comment by PaulSolDev** [comment]
Functional review complete. No issues identified.

Checked on standard desktop browsers, as well as as the following for mobile: Safari, Chrome, Firefox, and Edge.

**Pull request merged**